Project-Aurora/Project-Aurora/Devices/Ducky/DuckyDevice.cs

```diff
@@ -0,0 +1,360 @@
+            {DeviceKeys.NUM_LOCK, new int[] {6, 51}},
+            {DeviceKeys.NUM_SEVEN, new int[] {6, 54}},
+            {DeviceKeys.NUM_FOUR, new int[] {6, 57}},
+            {DeviceKeys.NUM_ONE, new int[] {6, 60}},
+            {DeviceKeys.NUM_ZERO, new int[] {6, 63}},
+
+            {DeviceKeys.VOLUME_MUTE, new int[] {7, 6}},
+            {DeviceKeys.NUM_SLASH, new int[] {7, 9}},
+            {DeviceKeys.NUM_EIGHT, new int[] {7, 12}},
+            {DeviceKeys.NUM_FIVE, new int[] {7, 15}},
+            {DeviceKeys.NUM_TWO, new int[] {7, 18}},
+          //{DeviceKeys., new int[] {7, 21}}, Nothing
+            {DeviceKeys.VOLUME_DOWN, new int[] {7, 24}},
+            {DeviceKeys.NUM_ASTERISK, new int[] {7, 27}},
+            {DeviceKeys.NUM_NINE, new int[] {7, 30}},
+            {DeviceKeys.NUM_SIX, new int[] {7, 33}},
+            {DeviceKeys.NUM_THREE, new int[] {7, 36}},
+            {DeviceKeys.NUM_PERIOD, new int[] {7, 39}},
+            {DeviceKeys.VOLUME_UP, new int[] {7, 42}},
+            {DeviceKeys.NUM_MINUS, new int[] {7, 45}},
+            {DeviceKeys.NUM_PLUS, new int[] {7, 48}},
+            //{DeviceKeys., new int[] {7, 51}},
+            //{DeviceKeys., new int[] {7, 54}}, Nothing for both.
+            {DeviceKeys.NUM_ENTER, new int[] {7, 57}}
+        };
+
+        private static string deviceName = "Ducky";
+        private bool isInitialized = false;
+        private long lastUpdateTime = 0;
+        private readonly Stopwatch watch = new Stopwatch();
+        private Color processedColor;
+        private int[] currentOffset;
+
+        private DuckyRGBAPI duckyAPI = new DuckyRGBAPI();
+        HidDevice shine7Keyboard;
+        HidStream packetStream;
+        byte[] colourMessage = new byte[640];
+        static byte[] startingPacket = { 0x56, 0x81, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x08, 0x00, 0x00, 0x00, 0xAA, 0xAA, 0xAA, 0xAA };
+        byte[] colourHeader = { 0x56, 0x83, 0x00 };
+
+        public VariableRegistry GetRegisteredVariables()
+        {
+            return new VariableRegistry();
+        }
+
+        public string GetDeviceName()
+        {
+            return deviceName;
+        }
+
+        public string GetDeviceDetails()
+        {
+            return deviceName + ": " + (isInitialized ? "Connected" : "Not initialized");
+        }
+
+        public string GetDeviceUpdatePerformance()
+        {
+            return (isInitialized ? lastUpdateTime + " ms" : "");
+        }
+
+        public bool Initialize()
+        {
+            //Sets the initialize colour change packet
+            startingPacket.CopyTo(colourMessage, Packet(0) + 1);
+            //Headers for each colour packet
+            for (byte i = 0; i < 8; i++)
+            {
+                colourHeader[2] = i;
+                colourHeader.CopyTo(colourMessage, Packet(i + 1) + 1);
+            }
+            //First colour packet has extra data
+            colourMessage[Packet(1) + 5] = 0x01;
```

Comment:
i'd maybe throw these magic bytes in a static array called `firstColorPacketData` or something and then Array.Copy it to `colorMessage`.
are the ones you don't set supposed to be zero?